src/api/soap.ts

```diff
@@ -240,6 +243,24 @@ export class SoapApi<S extends Schema> {
     };
     return this._invoke('delete', args, schema);
   }
+
+  /**
+   * Undelete records from the recycle bin immediately
+   */
+  undelete(ids: string | string[]): Promise<UndeleteResult[]>;
+  undelete(id: string): Promise<UndeleteResult>;
+  undelete(ids: string | string[]): Promise<UndeleteResult | UndeleteResult[]>;
+  undelete(ids: string | string[]) {
+    const schema = Array.isArray(ids)
+      ? [ApiSchemas.UndeleteResult]
+      : ApiSchemas.UndeleteResult;
+    const args = {
+      '@xmlns': 'urn:partner.soap.sforce.com',
+      '@xmlns:ns1': 'sobject.partner.soap.sforce.com',
+      'ns1:ids': ids,
```

Comment:
have you tried undeleting a single ID (as a string, not a 1-item array)? docs mention an array only:
https://developer.salesforce.com/docs/atlas.en-us.api.meta/api/sforce_api_calls_undelete.htm

Reply:
I have not try but you're right.
I'll try and I will fix if necessary

Reply:
As doc mention only array I simplified the code to follow the doc